File: Assessment Time/Objective-C/ViewController.m
/*
See LICENSE folder for this sample’s licensing information.

Abstract:
A web view.
*/
#import "ViewController.h"
#import <WebKit/WebKit.h>

@interface ViewController() <WKNavigationDelegate>
@property (weak) IBOutlet NSTextField *urlField;
@property (weak) IBOutlet WKWebView *webView;

@end
@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.webView.navigationDelegate = self;
    self.urlField.stringValue = @"https://www.apple.com";
    [self loadWebPage:self.urlField];
}


- (void)setRepresentedObject:(id)representedObject {
    [super setRepresentedObject:representedObject];

    // Update the view if already loaded.
}
- (IBAction)loadWebPage:(id)sender {
    // Make sure the url specifies the https scheme.
    if ([self.urlField.stringValue hasPrefix:@"http://"]) {
        self.urlField.stringValue = [self.urlField.stringValue stringByReplacingCharactersInRange:NSMakeRange(0, 4) withString:@"https"];
    } else if (![self.urlField.stringValue hasPrefix:@"https://"]) {
        self.urlField.stringValue = [@"https://" stringByAppendingString:self.urlField.stringValue];
    }
    
    NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:self.urlField.stringValue]];
    if ( request != NULL ) {
        [self.webView loadRequest:request];
    }
}

- (void)webView:(WKWebView *)webView didFinishNavigation:(WKNavigation *)navigation {
    if ( [webView URL] ) {
        self.urlField.stringValue = [[webView URL] absoluteString];
    }
}

@end

File: Assessment Time/Swift/ViewController.swift
/*
See LICENSE folder for this sample’s licensing information.

Abstract:
A web view.
*/

import Cocoa
import WebKit

@objc(ViewController)
class ViewController: NSViewController {
    
    @IBOutlet weak var urlField: NSTextField!
    @IBOutlet weak var webView: WKWebView!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        
        webView.navigationDelegate = self
        urlField.stringValue = "https://www.apple.com"
        loadWebPage(urlField)
        
    }
    
    @IBAction func loadWebPage(_ sender: NSTextField) {
        // Make sure the url specifies the https scheme.
        if urlField.stringValue.lowercased().hasPrefix("http://") {
            urlField.stringValue = "https" + urlField.stringValue.lowercased().dropFirst("http".count)
        } else if !urlField.stringValue.lowercased().hasPrefix("https://") {
            urlField.stringValue = "https://" + urlField.stringValue
        }

        // Fail silently if a URL can't be initialized.
        guard let url = URL(string: urlField.stringValue) else { return }
        
        let request = URLRequest(url: url)
        
        webView.load(request)
    }
}

extension ViewController: WKNavigationDelegate {
    // As the user navigates from one page to another, keep the URL text field up to date.
    func webView(_ webView: WKWebView, didFinish navigation: WKNavigation!) {
        guard let url = webView.url else { return }
        urlField.stringValue = "\(url)"
    }
}
